src/datascience-ui/ipywidgets/kernel/index.ts

```diff
@@ -218,14 +222,18 @@ function convertVSCodeOutputToExecuteResultOrDisplayData(
     disposeOutput
 };
 
+let capturedContext: KernelMessagingApi | undefined;
 // To ensure we initialize after the other scripts, wait for them.
-function attemptInitialize() {
+function attemptInitialize(context?: KernelMessagingApi) {
+    capturedContext = capturedContext || context;
+    console.log('Attempt Initialize IpyWidgets kernel.js', context);
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     if ((window as any).vscIPyWidgets) {
         logMessage('IPyWidget kernel initializing...');
-        initialize();
+        initialize(capturedContext);
     } else {
         setTimeout(attemptInitialize, 100);
     }
 }
-attemptInitialize();
+
+export const activate = attemptInitialize;
```

Comment:
This is new, the way VS Code works is weird.
If the text `export ... activate` exists anywhere in the JS code, then VS Code will load the JS file as a ES module & call the activate (at that point the class `acquireVSCodeAPI` doesn't exist in the global context).

Note: We weren't expoting the active method, but somewhere in the code that text ends up in the JS file and VS Code tries to load this JS as an ES module and fails.
Solution: Ensure we always setup this script file to be loaded as an es module.
This also changes how the messaging api is handled (that's now an argument in the activate method).

Reply:
Will discuss in standup